public/js/components/Editor.js

```diff
@@ -1,30 +1,38 @@
 const React = require("react");
+const ReactDOM = require("react-dom");
 const ImPropTypes = require("react-immutable-proptypes");
 const { bindActionCreators } = require("redux");
 const { connect } = require("react-redux");
-const CodeMirror = require("codemirror");
+const SourceEditor = require("../utils/source-editor");
 const { DOM: dom, PropTypes } = React;
-const { debugGlobal } = require("../utils/debug");
-const { isFirefox } = require("../feature");
 
 const {
   getSourceText, getBreakpointsForSource,
   getSelectedSource, getSelectedFrame
 } = require("../selectors");
 const { makeLocationId } = require("../reducers/breakpoints");
 const actions = require("../actions");
-const { alignLine, onWheel, resizeBreakpointGutter } = require("../utils/editor");
 const Breakpoint = React.createFactory(require("./EditorBreakpoint"));
 
-require("codemirror/lib/codemirror.css");
 require("./Editor.css");
-require("codemirror/mode/javascript/javascript");
-require("../lib/codemirror.css");
 
 function isSourceForFrame(source, frame) {
   return frame && frame.location.sourceId === source.get("id");
 }
 
+/**
+ * Forces the breakpoint gutter to be the same size as the line
+ * numbers gutter. Editor CSS will absolutely position the gutter
+ * beneath the line numbers. This makes it easy to be flexible with
+ * how we overlay breakpoints.
+ */
+function resizeBreakpointGutter(editor) {
+  const gutters = editor.display.gutters;
+  const lineNumbers = gutters.querySelector(".CodeMirror-linenumbers");
+  const breakpoints = gutters.querySelector(".breakpoints");
+  breakpoints.style.width = lineNumbers.clientWidth + "px";
+}
```

Comment:
If you think there will be other editor utils, it might be nice to bring back `utils/editor.js`